### ISSUE TITLE
Add protocol to supply ZN Coin mining.

### ISSUE THREAD

**Issue by HanEightTurtle** [HelloZeroNet/ZeroNet#2848]
I hope there will be a corresponding coin, such as ZN coin, to incentivize nodes involved in building sites or participating in caching. Additionally, a distributed DNS server can be set up, allowing site owners to spend ZN coin to purchase domain names through the protocol.
  In my opinion, ZeroNet is the true Web3.

**Comment by caryoscelus** [comment]
there are (were?) a lot of people preferring 0net precisely because of no crypto associated with it. my personal stance is that no blockchain should be mandatory to any part of it (including no "distributed" "dns") because a) commercial incentives are attracting bad actors and b) the concept of consensus is a different form of centralization

that said, i don't see anything wrong with building a layer of incentivizing storage nodes.. if anyone can make it abuse-proof

anyhow, this repo is dead, see fork(s) for more alive version(s)

**Comment by mx5kevin** [comment]
The download of files should be proportional to the size of the seeding, or helping in hosting,  or pay for it. The same filter with user registration and commenting. With this, a working business model could be built and unwanted users could be filtered at the same time.

**Comment by wolfitdm** [comment]
> there are (were?) a lot of people preferring 0net precisely because of no crypto associated with it. my personal stance is that no blockchain should be mandatory to any part of it (including no "distributed" "dns") because a) commercial incentives are attracting bad actors and b) the concept of consensus is a different form of centralization
> 
> that said, i don't see anything wrong with building a layer of incentivizing storage nodes.. if anyone can make it abuse-proof
> 
> anyhow, this repo is dead, see fork(s) for more alive version(s)

yes you are right, this repo is dead, and yes i want do that too... but for now i put that here...

I see the future more in a pluggable blockchain....

So for ZeroID, I see adaptations like this as a plugin that make sense...

python implementations:

https://github.com/codebox/blockchain
https://github.com/obulpathi/bitcoinpy/blob/master/bitcoinpy/bitcoin.py
https://github.com/subhankardas2000/Blockchain-with-Python

and it exists over 10000000000000000000000000 blockchain projects in html/javascript/python and so on...

you can use that for zeronet.... without to get in "license issues"

In the sense of a kind of "decentralized verification" that replaces Namecoin and where the coins are worth nothing...

Where ZeroMoney only serves to concentrate fully on the utility value...

It doesn't have to be a "huge blockchain" either.

Because what doesn't exist yet is the ability to decide whether you want a blockchain or not....

At the moment, it's always been dependent....

Take any coin...

To have the wallet, you need a blockchain...

Let's imagine you could now say, hmm, today I have a blockchain and tomorrow none???

A kind of pluggable blockchain... That doesn't exist yet and this enormous independence from anything is exactly what makes ZeroNet so special...

ZeroName for example... you are not dependent on a DNS, but you have the option of being dependent on a DNS... through the "disabled plugins"...


@caryoscelus 
But yes, I agree with you that it shouldn't be mandatory....

A kind of ZeroVerify or ZeroMoneyPlugin, which is based on PeerMessage, could be in the spirit of ZeroNet...

You could use it to exchange messages and certificates.,...

There are also very lightweight blockchains for Python, which are more in the spirit of Zeronet...

As I said, it doesn't have to be a cryptocurrency...

There is a difference between blockchain and cryptocurrency...

As I said, with a kind of zero money plugin where you farm an average of 10000000000, it's worth nothing, but it's ideal for keeping memory consumption low and also keeping bad actors away...

So with the right implementation, I see it as useful....

and also can be used for name operations...

perfect and pluggable........

not a "fixed non zero implementation" 

Have a nice day....

I am go sleeping now....:)
